### PR TITLE
Fix for Patient Portal Responsiveness (#2793)

### PR DESCRIPTION
**Pull request by tywrenn**
This fixes a ton of responsive design flaws in the Patient Portal.
This is change 1 of 2 for #2793. The 2nd change will fix the register.php area as well as fixes any errors done on my part (now available).

Changelog of Overall Changes:
- CSS beautified base.css and register.css
- Fixed a ton of Bootstrap code
- Added dynamic padding to wrapper class that changes based upon device type
- Fixed login logo to be under Bootstrap responsive image classes
- Added dynamic sizing to login logo
- Replaced table in reset password area with Bootstrap CSS
- Removed a ton of col-xs-12 as it is not needed at all in this CSS
- Removed repetitive class text-center declarations
- Removed inline CSS code and placed into external CSS
- Changed login background color and wrapper background color to match style on main login page
- Other fixes

Please let me know what should be fixed when I commit the 2nd part of this fix.

### CODE REVIEW

**Comment by sjpadgett** [comment]
Great, finally someone else takes an interest in portal. Can't wait to pull down and test and will do that in next couple hours.

**Comment by sjpadgett** [comment]
Very nice so far. I await next pass.

**Comment by tywrenn** [comment]
Correct me if i'm wrong - it looks like I need to convert the file to linux headers to fix travis ci?

**Comment by sjpadgett** [comment]
I run them from a git bash on windows. You can also do:
php /xampp/htdocs/phpcbf.phar -n --standard=ci/phpcs.xml portal/index.php

**Comment by stephenwaite** [comment]
for linux travis, outside of the git repo will run
```
composer global require "squizlabs/php_codesniffer=3.*"
```
followed by
```
$HOME/.composer/vendor/bin/phpcbf -p -n --extensions=php,inc --standard=ci/phpcs.xml <directory to codesniff>
```

**Comment by bradymiller** [comment]
hi @tywrenn ,
here's the travis issues:
```
FILE: /home/travis/build/openemr/openemr/portal/index.php
----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 1 LINE
----------------------------------------------------------------------
 262 | ERROR | [x] Each PHP statement must be on a line by itself
 262 | ERROR | [x] Each PHP statement must be on a line by itself
----------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY

```
(looks like having separate php lines of code on 1 line (ie. semicolons) is causing the issue)

**Comment by tywrenn** [comment]
This is commit 2 of 2 for this request that focuses more on the register.php file:

Overall Changelog:
- Removes more of the outdated col-xs-12
- Fixes the form tag for first name, last name, birth date, and email address as the required tag was not properly coded in
- Sets the iFrame registration area to actually span the entire height of the content of the frame to allow easier use (required modification of patient list view to do so)
- Fixes an index.php issue where the meta tag wasn't set correctly so no mobile devices could actually view it correctly
- Adds "Patient Registration" to the top of the registration page
- Other fixes

Finally fixed Travis CI :)

**Comment by sjpadgett** [comment]
Very good. I'll have to look closer at this tomorrow as i'm too tired tonight. Cursory look looks good.
This was originally written against bootstrap 2 and got carried forward over time. 
What you've done so far will make it much easier to go to bootstrap 4. So thanks, this needed to be done.
You're pretty good at this. Want to do another project?

**Comment by tywrenn** [comment]
Sure :)

**Comment by sjpadgett** [comment]
Okay, let me think of something interesting for you to tackle. Get back to you in a.m.

File: portal/index.php [hunk 1]
```diff
@@ -463,7 +431,7 @@ function validate_new_pass() {
                 <?php $datetimepicker_timepicker = false; ?>
                 <?php $datetimepicker_showseconds = false; ?>
                 <?php $datetimepicker_formatInput = false; ?>
-                <?php require($GLOBALS['srcdir'] . '/js/xl/jquery-datetimepicker-2-5-4.js.php'); ?>
+                <?php include $GLOBALS['srcdir'] . '/js/xl/jquery-datetimepicker-2-5-4.js.php'; ?>
```
Comment (bradymiller) [inline portal/index.php]: why change this to an include?

**Comment by bradymiller** [comment]
hi @tywrenn , Just did a code review (very nice stuff!) and just noted 1 issue (see my post). I didn't test this and will let @sjpadgett have the honors. -brady

File: portal/index.php (hunk 1 above)
Comment (sjpadgett) [reply portal/index.php]: Agree. I've found the best use of include is if you want to handle problems with include loading in code. Sometimes you don't want your app to stop cold as a require will do but seek other ways to continue app flow. In this case, require is appropriate.

Comment (tywrenn) [reply portal/index.php]: What's weird is I didn't actually modify this line - the code linter did this 😂

Comment (stephenwaite) [reply portal/index.php]: that is weird, pretty sure it shouldn't be doing that
what codesniffer commands did you use?

Comment (tywrenn) [reply portal/index.php]: Just edited it back now and it's not spitting out any errors - that was weird, probably executed the command wrong at one point so let me know if there are any other errors before I commit my last commit for this pull request.

Comment (sjpadgett) [reply portal/index.php]: Nope you're good. Once travis is finish with your commit, i'll merge to master. I am going to be working with appointments next for handling recurring events.

**Comment by tywrenn** [comment]
Done :)

**Comment by sjpadgett** [comment]
Congrats Tyler on your first PR and peer review on OpenEMR. Nicely done and I think you've impressed some of us. Hopefully this is the first of many you will contribute to the community.

I just did some work in our calendar. While I enjoy doing the architectural, styling, not so much.
Want to take a look at the calendar? You can start by reviewing my latest PR's for a starting place as to where code is located and what i've done thus far. What do you think?

**Comment by tywrenn** [comment]
Sure i'll take a look - also, we might want to start thinking about transitioning to Bootstrap 4 - the revision that is currently on OpenEMR dates back to 2015 😂 I think I might be able to help out with that transition when you start thinking about dealing with that just let me know

**Comment by sjpadgett** [comment]
Now is as good a time as any. I'd also bet you would get some support/advice for some others whom have wanted bootstrap 4.
I'm unaware of anywhere we are using bootstrap 4 so, unclear why it's in dependencies. I'm sure someone will correct me on this. Perhaps @mdsupport used somewhere or has a suggestion where we might start.
Meantime, will open an issue to work from and say goodbye to this PR.

**Comment by bradymiller** [comment]
Getting to bootstrap 4 would be awesome. A possible sticking point is RTL support (OpenEMR currently has strong RTL support that requires a bootstrap rtl package).

**Comment by sjpadgett** [comment]
Aw yes, I forgot RTL. I think it was dropped in BS4 but will research this for a solution.

**Comment by bradymiller** [comment]
BS3 didn't officially support RTL, but there was this library here that we use that supported the RTL for BS3:
https://github.com/morteza/bootstrap-rtl
We may be in luck:
https://github.com/MahdiMajidzadeh/bootstrap-v4-rtl

**Comment by tywrenn** [comment]
RTL should only be a temporary solution - anyways moving this convo to issue #2807

**Pull request merged**